internal/testutil/acc/config_formatter.go

```diff
@@ -50,3 +57,183 @@ func sortStringMapKeys(m map[string]string) []string {
 	sort.Strings(keys)
 	return keys
 }
+func sortStringMapKeysAny(m map[string]any) []string {
+	keys := make([]string, 0, len(m))
+	for k := range m {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	return keys
+}
+
+var matchFirstCap = regexp.MustCompile("(.)([A-Z][a-z]+)")
+var matchAllCap = regexp.MustCompile("([a-z0-9])([A-Z])")
+
+func ToSnakeCase(str string) string {
+	snake := matchFirstCap.ReplaceAllString(str, "${1}_${2}")
+	snake = matchAllCap.ReplaceAllString(snake, "${1}_${2}")
+	return strings.ToLower(snake)
+}
+
+func ClusterResourceHcl(projectID string, req *ClusterRequest, specs []admin.ReplicationSpec) (configStr, clusterName string, err error) {
+	if len(specs) == 0 {
+		specs = append(specs, ReplicationSpec(&ReplicationSpecRequest{}))
+	}
+	if req == nil {
+		req = new(ClusterRequest)
+	}
+	clusterName = req.ClusterNameExplicit
+	if clusterName == "" {
+		clusterName = RandomClusterName()
+	}
+	clusterTypeStr := "REPLICASET"
```

Comment:
what about SHARDED type?

Reply:
It wasn't supported before.
We can add it on a "request-basis"